apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -37,12 +37,399 @@ type DatadogAgentSpec struct {
+
+	// LogCollection configuration.
+	LogCollection *LogCollectionFeatureConfig `json:"logCollection,omitempty"`
+	// ProcessCollection configuration.
+	ProcessCollection *ProcessCollectionFeatureConfig `json:"processCollection,omitempty"`
+	// ContainerCollection configuration.
+	ContainerCollection *ContainerCollectionFeatureConfig `json:"containerCollection,omitempty"`
+	// APM (Application Performance Monitoring) configuration.
+	APM *APMFeatureConfig `json:"apm,omitempty"`
+	// CSPM (Cloud Security Posture Management) configuration.
+	CSPM *CSPMFeatureConfig `json:"cspm,omitempty"`
+	// CWS (Cloud Workload Security) configuration.
+	CWS *CWSFeatureConfig `json:"cws,omitempty"`
+	// NPM (Network Performance Monitoring) configuration.
+	NPM *NPMFeatureConfig `json:"npm,omitempty"`
+	// USM (Universal Service Monitoring) configuration.
+	USM *USMFeatureConfig `json:"usm,omitempty"`
+
+	// Cluster-level features
+
+	// OrchestratorExplorer check configuration.
+	OrchestratorExplorer *OrchestratorExplorerFeatureConfig `json:"orchestratorExplorer,omitempty"`
+	// KubeStateMetricsCore check configuration.
+	KubeStateMetricsCore *KubeStateMetricsCoreFeatureConfig `json:"kubeStateMetricsCore,omitempty"`
+	// AdmissionController configuration.
+	AdmissionController *AdmissionControllerFeatureConfig `json:"admissionController,omitempty"`
+	// ExternalMetricsServer configuration.
+	ExternalMetricsServer *ExternalMetricsServerFeatureConfig `json:"externalMetricsServer,omitempty"`
+	// ClusterChecksRunner configuration.
+	ClusterChecksRunner *ClusterChecksRunnerFeatureConfig `json:"clusterChecksRunner,omitempty"`
+	// PrometheusScrape configuration.
+	PrometheusScrape *PrometheusScrapeFeatureConfig `json:"prometheusScrape,omitempty"`
+	// DatadogMonitor configuration.
+	DatadogMonitor *DatadogMonitorFeatureConfig `json:"datadogMonitor,omitempty"`
+}
+
+// Configuration structs for each feature in DatadogFeatures. All parameters are optional and have default values when necessary.
+// Note: configuration in DatadogAgentSpec.Override takes precedence.
+
+// APMFeatureConfig contains APM (Application Performance Monitoring) configuration.
+// APM runs in the Trace Agent.
+type APMFeatureConfig struct {
+	// Enabled enables Application Performance Monitoring.
+	// Default: false
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// HostPortConfig contains host port configuration.
+	// Enabled Default: false
+	// Port Default: 8126
+	// +optional
+	HostPortConfig *HostPortConfig `json:"hostPortConfig,omitempty"`
+
+	// UnixDomainSocketConfig contains socket configuration.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm#agent-environment-variables
+	// Enabled Default: true
+	// Path Default: `/var/run/datadog/apm.socket`
+	// +optional
+	UnixDomainSocketConfig *UnixDomainSocketConfig `json:"unixDomainSocket,omitempty"`
+}
+
+// HostPortConfig contains host port configuration.
+type HostPortConfig struct {
```

Comment:
nit: it would be nice to not mix the minor struct like `HostPortConfig` in the middle of the features one

Reply:
agreed, will move